gateway-server/src/main/java/org/apache/knox/gateway/webshell/JSchConnectInfo.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.knox.gateway.webshell;
+
+import com.jcraft.jsch.Channel;
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Session;
+import org.eclipse.jetty.io.RuntimeIOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+
+/**
+ * data structure to store a connection session
+ */
+public class JSchConnectInfo extends ConnectInfo {
+
+    private JSch jsch;
+    private Channel channel;
+    private Session jschSession;
+    private static final Logger LOG = LoggerFactory.getLogger(JSchConnectInfo.class);
+    public JSchConnectInfo(){
+        try {
+            jsch = new JSch();
+            Properties config = new Properties();
+            config.put("StrictHostKeyChecking", "no");
+            jschSession = jsch.getSession("knoxui", "localhost", 22);
+            jschSession.setConfig(config);
+            jschSession.setPassword("knoxui");
+        }catch (JSchException e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void connect(String username){
+        try {
+            jschSession.connect(30000);
+            channel = jschSession.openChannel("shell");
+            inputStream = channel.getInputStream();
+            outputStream = channel.getOutputStream();
+            channel.connect(30000);
+            String sudoCmd = "exec sudo -u "+ username + " -H bash -i\ncd $HOME\nwhoami\n";
+            outputStream.write(sudoCmd.getBytes(StandardCharsets.UTF_8));
+            outputStream.flush();
+            checkConnection(username);
+        }  catch (JSchException|IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void checkConnection(String username) throws IOException{
```

Comment:
I am not sure what this method is doing, is this validating connection based on just username?

Reply:
i should organize/document this part better. 
I want to make sure bash is running as the target user before passing the terminal output to the browser terminal. 
so in connect() i sent command `whoami` to bash and it should return the targeted username. 
i haven't thought of a better way to do that.